replica/database.cc

```diff
@@ -2099,7 +2094,7 @@ future<> database::apply_hint(schema_ptr s, const frozen_mutation& m, tracing::t
     if (dblog.is_enabled(logging::log_level::trace)) {
         dblog.trace("apply hint {}", m.pretty_printer(s));
     }
-    if (!s->is_synced()) {
+    if (!(s->is_synced() || s->get_reversed()->is_synced())) {
```

Comment:
Also, it deserves an explanation how is_synced can be different on the schema and its reversed schema.

Maybe the right direction is to make sure they are in sync with each other if that's correct, and then you won't need this change.

If we do, then better print also the reversed schema version in the error message since if we get the reversed table schema here, I'm not sure its (reversed) version would be helpful, as far as finding it in the logs.